runtime/lua/vim/_system.lua

```diff
@@ -90,19 +90,23 @@ end
 
 local MAX_TIMEOUT = 2 ^ 31
 
---- @param timeout? integer
+--- @param timeout? integer timeout (in milliseconds), must be >= 0.
```

Comment:
Should be strictly greater than 0 right? Since we apply `math.max` just below this, if a user puts `timeout = 0` they will be confused that the actual timeout is `1`, even though the docs say `0` is supported.

```suggestion
--- @param timeout? integer timeout (in milliseconds), must be > 0.
```

Reply:
I agree that `wait(0)` or `wait(0.1)` (a possible common mistake) does not do anything useful or semantically meaningful -- it's to immediately terminate a process. Alternatively, we can choose to disallow `timeout < 1` (including values like 0 or 0.5), raising explicit errors.